base/logging.jl

```diff
@@ -44,14 +45,18 @@ function handle_message end
 
 Return true when `logger` accepts a message at `level`, generated for
 `_module`, `group` and with unique log identifier `id`.
+
+For very early filtering of custom log levels, users may override
+`shouldlog(level)`.
```

Comment:
Is there a practical example other than `severity(level) >= _min_enabled_severity[]`?

Reply:
Never mind. I guess one application might be some kind of "zero-cost `@debug`" like ToggleableAsserts.jl https://discourse.julialang.org/t/ann-toggleableasserts-jl/31291

Reply:
You got it, this is exactly what I had in mind!

Reply:
By the way, this has been re-invented multiple times, for example https://discourse.julialang.org/t/defensive-programming-assert/8383/11?u=chris_foster :-D